File: python/adivina.py
'''
Juego de adivina el numero
'''

from random import randint
from utilidades import encabezado, linea, stringToNumber, readString

def main () -> None:
#definir variables
 secreto: int = randint(0, 50)
 numero: int = 0
 alias: str = ''

 encabezado('Juego de adivina el numero', 32)
 alias = readString('Agrega tu alias', 'No puede estar vacia')
 print(f"Hola, {alias}")
 linea(32)
 jugar(secreto, numero, alias)
 
def jugar(secreto: int, numero: int, alias: str) -> None:
  intentos: int = 0
  msj: str = ''
  while numero != secreto:
    if intentos == 6:
      print(f'{alias}, Perdiste, ya no te quedan mas intentos, el numero era {secreto}')
      break
    numero = stringToNumber('Ingresa tu numero: ', int)
    intentos += 1
    print(f'llevas {intentos} intentos')
    if numero > secreto:
      msj = 'tu numero es mayor a secreto...'
      print(f'{alias.upper()}, {msj}')
    elif numero < secreto:
      msj = 'tu numero es menor a secreto...'
      print(f'{alias.upper()}, {msj}')
    else:
      msj = 'ganaste'
      linea(45)
      print(f'{alias.upper()}, {msj.upper()}')
      linea(45)
      
main()

File: dart/lib/adivina.dart
import 'dart:math';
import 'package:colorize/colorize.dart';
import 'util.dart' as util;

// VARIABLES
final random = Random();
final _numberRandom = random.nextInt(100) + 1;
int _attemps = 7;

void main() {
  util.encabezado('ADIVINA EL NUMERO DEL 1 AL 100', 32);
  final alias = util.leerString('AGREGA TU USERNAME: ');
  jugar(alias);
}

//FUNCION PARA JUGAR
void jugar(String alias) {
   while (true) {
  final respuesta = util.leerString('AGREGA UN NUMERO: ');
  final diferencia = (int.parse(respuesta) - _numberRandom).abs();

    if (respuesta.isEmpty) {
      print('DEBES AGREGAR UN NUMERO VALIDO');
    }

    if (_attemps == 1) {
      print(Colorize('${alias.toUpperCase()} PERDISTE, EL NUMERO ERA $_numberRandom').bgRed());
      break;
    }
  
    if (int.tryParse(respuesta) == _numberRandom) {
      print(Colorize('${alias.toUpperCase()} GANASTE').green());
      break;
    } else if (int.tryParse(respuesta)! > _numberRandom) {
      print('MAS ABAJO');
      _attemps--;
      print('TE QUEDAN $_attemps intentos');
    } else if (int.tryParse(respuesta)! < _numberRandom) {
      print('MAS ARRIBA');
      _attemps--;
      print('TE QUEDAN $_attemps intentos');
    }else{
      print('PERDISTE');
      break;
    }
  
    if (diferencia <= 5) {
      print(Colorize('CALIENTE').red());
    } else if (diferencia <= 10) {
      print(Colorize('TIBIO').lightRed());
    } else {
      print(Colorize('FRIO').lightBlue());
    }
  }
}
